app/src/main/java/com/marverenic/music/player/PlayerService.java

```diff
@@ -497,6 +500,26 @@ public void setQueue(List<Song> newQueue, int newPosition) throws RemoteExceptio
         }
 
         @Override
+        public void beginBigQueue() throws RemoteException {
+            mQueueTemp = new ArrayList<>();
+        }
+
+        @Override
+        public void sendQueueChunk(List<Song> chunk) throws RemoteException {
+            Preconditions.checkNotNull(mQueueTemp).addAll(chunk);
```

Comment:
`Preconditions` is in Dagger's internal package, so I'm kind of hesitant to use it here. I think we should just do something like:

```
if (mQueueTemp == null) {
    throw new IllegalStateException("Must call beginBigQueue() to start a transaction");
}
mQueueTemp.addAll(chunk)
```